app/api/v1/analytics.py

```
@@ -0,0 +1,509 @@
+from app.models.business_intelligence import Company, Opportunity
+from app.schemas.analytics import (
+    LeadScoreResponse, ROIMetricsResponse, PredictiveAnalyticsResponse,
+    CompetitiveIntelligenceResponse, CampaignAnalyticsResponse, BusinessMetricsResponse
+)
+
+router = APIRouter()
+
+@router.get("/advanced-overview", response_model=Dict[str, Any])
+async def get_advanced_analytics_overview(
+    time_range: str = Query("30d", regex="^(7d|30d|90d|1y)$"),
+    db: Session = Depends(get_db)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)